docs/design-survey-conditional-logic.md

```diff
@@ -0,0 +1,691 @@
+| `"{speakinglanguages} contains 'Spanish'"` | Evaluates to `true` if Spanish is selected. Other languages may or may not be selected. |
+| `"age({birthdate}) >= 21"` | Evaluates to `true` if the `age` function returns 21 or higher. |
+
+You should use different properties to specify the visibility of [questions](#question-visibility) and [items (choices, rows, columns)](#item-visibility-choices-columns-rows).
+
+### Question Visibility
+
+Assign Boolean expressions to the [visibleIf](https://surveyjs.io/Documentation/Library?id=Question#visibleIf), [enableIf](https://surveyjs.io/Documentation/Library?id=Question#enableIf), and [requiredIf](https://surveyjs.io/Documentation/Library?id=Question#requiredIf) properties of questions, panels, and pages. In the following example, the `visibleIf` property is used to hide the `drivers-license` question for respondents under 16 years old:
+
+```js
+const surveyJson = {
+  "elements": [{
+    "name": "birthdate"
+  }, {
+    "name": "drivers-license",
+    "title": "Have you got a driver's license?",
+    "visibleIf": "age({birthdate}) >= 16"
+  }]
+}
+```
+
+If you do not specify the `visibleIf`, `enableIf`, and `requiredIf` properties, an element's state depends on the [`isVisible`](/Documentation/Library?id=Question#isVisible), [`isReadOnly`](/Documentation/Library?id=Question#isReadOnly), and [`isRequired`](/Documentation/Library?id=Question#isRequired) properties. You can specify them at design time or use them to get or set the current state at runtime. If you set one of these properties for a panel or page, all nested questions inherit the setting.
+
+- [View the "Conditional Visibility" example](/Examples/Library?id=condition-kids)
+- [View the "Enable/Disable Elements" example](/Examples/Library?id=condition-enable-kids)
+
+### Item Visibility (Choices, Columns, Rows)
+
+SurveyJS allows you to control available choices, columns, and rows based on previous answers.
+
+#### Specify Visibility Conditions for Individual Items
+
+Individual items (choices, columns, rows) can be configured with objects. Each object can have a `visibleIf` property that accepts an expression. When the expression evaluates to `true`, the associated item becomes visible.
+
+In the following code, the `SMS` and `WhatsApp` choices are visible only if a user has entered their phone number:
+
+```js
+const surveyJson = {
+  "elements": [{
+    "name": "Contacts"
+    "choices": [
+      "Email",
+      { "value": "SMS", "visibleIf": "{phone} notempty" },
+      { "value": "WhatsApp", "visibleIf": "{phone} notempty" }
+    ]
+  },
+  // ...
+  ]
+}
+```
+
+[View example](https://surveyjs.io/Examples/Library?id=condition-choicesVisibleIf)
+
+This technique has one drawback: if a question contains many items, you have to copy the same expression into every item that should have dynamic visibility. If that is your case, use a technique described in the next topic.
```

Comment:
This technique has one drawback: if a question contains many items, you have to copy the same expression into every item that should have dynamic visibility. If that is your case, use the technique described in the next topic.